src/map/script.c

```diff
@@ -8616,6 +8616,58 @@ static BUILDIN(delitem2)
 	return false;
 }
 
+/**
+ * Deletes item at given index.
+ * delitem(<index>{, <account id>});
+ */
+static BUILDIN(delitemidx)
+{
+	struct map_session_data *sd;
+
+	if (script_hasdata(st, 3)) {
+		if ((sd = script->id2sd(st, script_getnum(st, 3))) == NULL) {
+			script_pushint(st, 0);
+			st->state = END;
+			return true;
+		}
+	} else {
+		if ((sd = script->rid2sd(st)) == NULL)
+			return true;
+	}
+
+	int i = script_getnum(st, 2);
+	if (i < 0 || i >= sd->status.inventorySize) {
+		ShowError("buildin_delitemidx: Index (%d) should be from 0-%d.\n", i, sd->status.inventorySize - 1);
+		st->state = END;
+		return false;
+	}
+
+	int nameid = sd->status.inventory[i].nameid;
+	struct item_data *item_data = itemdb->exists(nameid);
+	if (item_data == NULL) {
+		ShowError("buildin_delitemidx: Invalid Item ID (%d).\n", nameid);
+		st->state = END;
+		return false;
+	}
+
+	struct item it;
+	memset(&it, 0, sizeof(it));
+	it.nameid = nameid;
+
```

Comment:
see `delitem2` ...
```
	it.amount=script_getnum(st,3);
	it.identify=script_getnum(st,4);
	it.refine=script_getnum(st,5);
	it.attribute=script_getnum(st,6);
	it.card[0] = script_getnum(st, 7);
	it.card[1] = script_getnum(st, 8);
	it.card[2] = script_getnum(st, 9);
	it.card[3] = script_getnum(st, 10);
```
in other words, this only look for the ID, and might delete wrong item
and yeah, no random option check as well

you already saw the item drop from monster can drop random option PR already ...